core/src/navigation/ContextSwitcher.html

```diff
@@ -0,0 +1,184 @@
+
+        // options are loaded lazy by default
+        if (ContextSwitcherHelpers.isContextSwitcherDetailsView(currentPath, contextSwitcherConfig.parentNodePath)
+            || contextSwitcherConfig.lazyloadOptions === false) {
+          await this.fetchOptions();
+          await this.setSelectedContext(currentPath);
+        }
+      }
+
+      Routing.addRouteChangeListener(() => this.setSelectedContext(Routing.getCurrentPath()));
+    },
+    data() {
+      return {
+        dropDownStates: {},
+        contextSwitcherEnabled: false,
+        config: {},
+        actions: [],
+        options: null,
+        fallbackLabelResolver: null,
+        selectedLabel: null
+      };
+    },
+    computed: {
+      dropDownStatesNegated({ dropDownStates }) {
+        const negated = {};
+        for (const key in dropDownStates) {
+          if (dropDownStates.hasOwnProperty(key)) {
+            negated[key] = Helpers.getNegatedBoolString(dropDownStates[key]);
+          }
+        }
+        return negated;
+      }
+    },
+    helpers: {
+      getFallbackNodeName: (config, id) => {
+        return ContextSwitcherHelpers.getFallbackNodeName(config, id);
+      }
+    },
+    methods: {
+      async fetchOptions() {
+        let options = this.get().options;
+        if (!options || !options.length) {
+          const contextSwitcherOptions = await LuigiConfig.getConfigValueAsync('navigation.contextSwitcher.options');
+          options = await ContextSwitcherHelpers.generateSwitcherNav(this.get().config, contextSwitcherOptions);
+          this.set({options});
+        }
+      },
+      async setSelectedContext(currentPath) {
+        const config = this.get().config;
+        const options = this.get().options;
+        const parentNodePath = config.parentNodePath;
+        const fallbackLabelResolver = config.fallbackLabelResolver;
+        this.set({
+          selectedLabel: await ContextSwitcherHelpers.getSelectedLabel(currentPath, options, parentNodePath, fallbackLabelResolver)
+        })
+      },
+      async onActionClick(node) {
+        if (node.clickHandler) {
+          const result = await node.clickHandler(node);
+          // If the clickHandler returns true, open the link
+          if (!result) {
+            return;
+          }
+        }
+        this.goTo(node.link);
+      },
+      goTo(path) {
+        if (LuigiConfig.getConfigBooleanValue('routing.useHashRouting')) {
+          window.location.hash = path;
+        } else {
+          window.location.pathname = path;
+        }
+      },
+      toggleDropdownState(name) {
+        const dropDownStates = this.get().dropDownStates || {};
+        dropDownStates[name] = Helpers.getNegatedBoolString(dropDownStates[name]);
+        this.set({
+          dropDownStates
+        });
+
+        const isOpened = JSON.parse(dropDownStates[name]);
+        if (isOpened) {
+          this.fetchOptions();
+        }
+      }
+    }
+  };
+</script>
+
+<style type="text/scss">
```

Comment:
We should wrap a name inside a context switcher. Right now you can have a 100 letters long word that would break the top-navigation.

<img width="1581" alt="screen shot 2018-11-16 at 08 58 21" src="https://user-images.githubusercontent.com/40663133/48605710-d182e380-e97d-11e8-9554-edb92bac53fb.png">

Reply:
This will be handled in ticket https://github.com/kyma-project/luigi/issues/193

Reply:
I actually fixed it here: https://github.com/kyma-project/luigi/pull/190/commits/28627c90702323dbe8a7d6458b86daa6a2244cd4